network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py

```diff
@@ -72,10 +66,14 @@ def get_context(self, value, parent_context=None):
         if category:
             type = "category"
             query = slugify(category)
-            BlogPageCategory = apps.get_model('wagtailpages.BlogPageCategory')
-            category_object = BlogPageCategory.objects.get(name=category)
-            blogpage.extract_category_information(category_object.slug)
-            entries = blogpage.get_entries(context)
+            try:
+                # verify this category exists
+                category_object = BlogPageCategory.objects.get(name=category)
```

Comment:
This doesn't seem to work when category is `all`. It goes to `except` and hence does not return after result back.

Reply:
excellent point!